### ISSUE TITLE
[pickers] Duplicated id:s in DateRangePicker SectionList makes it difficult for screen readers to announce correctly

### ISSUE THREAD

**Issue by erikastiebel** [mui/mui-x#19274]
### Steps to reproduce

Steps:
1. Open this link to live example: [(required)](https://codesandbox.io/embed/xjj6jt)
2. Open developer tools and inspect the **from** month (MM) and look at the id for that span
3. Do the same on the **to** month. The id:s are the same.
 

### Current behavior

All elements for MM, DD and YYYY with class: MuiPickersSectionList-sectionContent MuiPickersInputBase-sectionContent have duplicated id:s. The year, month and day are given identical id values ​​so we have the same id on both year fields, both month fields and day fields. This can create problems and is strictly not allowed in the standard. You don't get information about from and to dates when you use a screen reader. More information is needed for screen reader users here to know what date you are changing or what is selected.

### Expected behavior

The IDs should be unique and it should be made clear to screen readers when the date is a from or to date when navigating with a screen reader. 

### Context

A better experience for screen reader users with unique id:s on the elements.

### Your environment

**Search keywords**: accessibility html-standard screen reader

**Order ID**: 114708

**Comment by michelengelen** [comment]
Hey @erikastiebel ... I did dig into this a bit and found a solution to it.
IMHO it feels not very elegant, so I'd like to have @flaviendelangle opinion about it before opening a PR.

```diff
--- a/packages/x-date-pickers-pro/src/internals/utils/date-fields-utils.ts
+++ b/packages/x-date-pickers-pro/src/internals/utils/date-fields-utils.ts
@@ -1,4 +1,5 @@
-import { FieldRangeSection } from '@mui/x-date-pickers/internals';
+import type { FieldRangeSection } from '@mui/x-date-pickers/internals';
+import type { FieldSection } from '@mui/x-date-pickers/models';

 export const splitDateRangeSections = (sections: FieldRangeSection[]) => {
   const startDateSections: FieldRangeSection[] = [];
@@ -29,3 +30,9 @@ export function getRangeFieldType(
   const fieldType = field.fieldType ?? 'multi-input';
   return fieldType;
 }
+
+export function isFieldRangeSection(
+  section: FieldSection | FieldRangeSection,
+): section is FieldRangeSection {
+  return 'dateName' in section;
+}
```

```diff
--- a/packages/x-date-pickers/src/internals/hooks/useField/useFieldSectionContentProps.ts
+++ b/packages/x-date-pickers/src/internals/hooks/useField/useFieldSectionContentProps.ts
@@ -1,6 +1,7 @@
 import * as React from 'react';
 import useEventCallback from '@mui/utils/useEventCallback';
 import useId from '@mui/utils/useId';
+import { isFieldRangeSection } from '@mui/x-date-pickers-pro/internals/utils/date-fields-utils.ts';
 import { UseFieldStateReturnValue } from './useFieldState';
 import { FieldSection, MuiPickersAdapter, PickerManager } from '../../../models';
 import { UseFieldDOMGetters, UseFieldInternalProps } from './useField.types';
@@ -167,6 +168,10 @@ export function useFieldSectionContentProps(
         format: section.format,
       });

+      const generatedId = isFieldRangeSection(section)
+        ? `${id}-${section.dateName}-${section.type}`
+        : `${id}-${section.type}`;
+
       return {
         // Event handlers
         onInput: handleInput,
@@ -176,7 +181,7 @@ export function useFieldSectionContentProps(
         onFocus: createFocusHandler(sectionIndex),

         // Aria attributes
-        'aria-labelledby': `${id}-${section.type}`,
+        'aria-labelledby': generatedId,
         'aria-readonly': readOnly,
         'aria-valuenow': getSectionValueNow(section, adapter),
         'aria-valuemin': sectionBoundaries.minimum,
@@ -191,7 +196,7 @@ export function useFieldSectionContentProps(
         tabIndex: isContainerEditable || sectionIndex > 0 ? -1 : 0,
         contentEditable: !isContainerEditable && !disabled && !readOnly,
         role: 'spinbutton',
-        id: `${id}-${section.type}`,
+        id: generatedId,
         'data-range-position': (section as FieldRangeSection).dateName || undefined,
         spellCheck: isEditable ? false : undefined,
         autoCapitalize: isEditable ? 'off' : undefined,
@@ -230,7 +235,7 @@ interface UseFieldSectionContentPropsParameters {
 }

 type UseFieldSectionContentPropsReturnValue = (
-  section: FieldSection,
+  section: FieldSection | FieldRangeSection,
   sectionIndex: number,
 ) => PickersSectionElement['content'];
```

**Comment by flaviendelangle** [comment]
OK we should definitely not have duplicated ids...
Am I missing something or is this id used by the `aria-labelledby` on the exact same element the `id` is being defined on?
Shouldn't it work without this id at all?
And if it doesn't, maybe we can just add the section index (instead of the section date name) and call it a day. The section order can't change anyway.

**Comment by michelengelen** [comment]
Using the section index would work as well ... i just used the `dateName` for semantics. But I agree that it adds a bit of boilerplate.

and yes, now that you mention it: the `aria-labelledby` does not make much sense here

**Comment by flaviendelangle** [comment]
You could even remove `section.type` from the ID by the way 😆 

If you want to remove `aria-labelledby`, check with @LukasTy and test it before, we might be missing some black magic 👀

**Comment by erikastiebel** [comment]
Thanks for digging in to this issue 😊

If it were possible to expand the translation keys used in the aria-label to include yearFrom and yearTo, monthFrom and monthTo, and dayFrom and dayTo, that would be a neat solution to make it clearer for the screen reader.

**Comment by michelengelen** [comment]
> You could even remove `section.type` from the ID by the way 😆

true ... I would keep it in though to give a bit of context. For pure uniqueness of the `id` this would be irrelevant though. I agree!

> If you want to remove `aria-labelledby`, check with [@LukasTy](https://github.com/LukasTy) and test it before, we might be missing some black magic 👀

Yes, lets wait for his opinion on this as well.
